### ISSUE TITLE
Dropout behaves differently on different devices

### ISSUE THREAD

**Issue by yezhejack** [pytorch/pytorch#25690]
I have two machines. One has a P100 and another has 4 K80s.
I found the following code output differently on the two machines.
It means seeding doesn't make the output of dropout reproducible on different devices. 
Is it by design?
If we can not make the output reproducible on different devices, it may be a trouble of comparing different models.
```
#!/bin/bash python
import torch
import numpy as np 
import random 
seed = 0
torch.manual_seed(seed)
torch.cuda.manual_seed(seed)
torch.cuda.manual_seed_all(seed)  
np.random.seed(seed)  
random.seed(seed)  
torch.manual_seed(seed)
torch.backends.cudnn.benchmark = False
torch.backends.cudnn.deterministic = True
a = torch.ones(1000,1000).to('cuda:0')
dropout = torch.nn.Dropout(0.5)
b = dropout(a)
print(torch.sum(torch.abs(b)))
```
The output of P100 machine is always
`
tensor(999826., device='cuda:0')
`
The output of K80 machine is always
`
tensor(999100., device='cuda:0')
`

**Comment by soumith** [comment]
we dont guarantee that the RNG outputs the same sequence across different GPU models.

The guarantees of RNG are in https://pytorch.org/docs/stable/notes/randomness.html?highlight=deterministic

**Comment by yezhejack** [comment]
Is it possible to make the RNG output same sequence across different GPU. The performance (metrics) of my generation model may change more than 20% on different GPUs.

**Comment by soumith** [comment]
> Is it possible to make the RNG output same sequence across different GPU. 

No it is not in our plans. We take RNG from CUDA API, and CUDA has made the decision of not giving same random sequences across GPUs.
